prediction-polls/backend/src/routines/pollCloseRoutine.js

```diff
@@ -0,0 +1,257 @@
+
+                // Sort float_answers array in ascending order
+                float_answers.sort((a, b) => a - b);
+                const correctAnswer = findMedian(float_answers)
+
+                await pollService.awardWinnersContinuousPoll(pollObject,correctAnswer,"float")
+
+                await modService.awardJuryContinuousPoll(pollObject,correctAnswer,"float")
+            }
+            else{
+                const date_answers = mod_requests.map(mod_request => (mod_request.date_value));
+
+                // Sort float_answers array in ascending order
+                date_answers.sort((a, b) => a - b);
+                const correctAnswer = findMedianDate(date_answers)
+
+                await pollService.awardWinnersContinuousPoll(pollObject,correctAnswer,"date")
+
+                await modService.awardJuryContinuousPoll(pollObject,correctAnswer,"date")
+            }
+            await modDB.deleteModRequestsforPollClose(pollObject.id)
+            await pollDB.finalizePoll(pollObject.id)
+        }
+        return {status:"success"}
+    }catch (error) {
+        console.error('Close Poll Routine: Database Error');
+        throw {error: errorCodes.DATABASE_ERROR};
+    }
+}
+
+async function checkLastGatheringTime(pollObject){
+
+    const lastGatheringTime = moment(pollObject.lastJuryGathering);
+    const overTime  = moment().diff(lastGatheringTime, 'hours');
+
+    // If overTime is bigger than reward hours, do another gathering
+    return overTime >= jury_reward_incerase_time_hours;
+}
+
+function findMedian(arr) {
+    const length = arr.length;
+
+    if (length % 2 === 0) {
+        // If the array has an even number of elements, average the middle two
+        const middle1 = arr[length / 2 - 1];
+        const middle2 = arr[length / 2];
+        return (middle1 + middle2) / 2;
+    } else {
+        // If the array has an odd number of elements, return the middle one
+        return arr[Math.floor(length / 2)];
+    }
+}
+
+function findMedianDate(arr) {
+    const length = arr.length;
+
+    if (length % 2 === 0) {
+        // If the array has an even number of elements, average the middle two
+        const middle1 = arr[length / 2 - 1];
+        const middle2 = arr[length / 2];
+        const averageMilliseconds = (middle1.getTime() + middle2.getTime()) / 2;
+        return new Date(averageMilliseconds);
+    } else {
+        // If the array has an odd number of elements, return the middle one
+        return arr[Math.floor(length / 2)];
+    }
+}
+
+async function updateJuryRewardForPoll(pollObject){
+    
+    const new_jury_reward = pollObject.juryReward * jury_reward_increase_constant;
+
+    await pollDB.updateJuryReward(pollObject.id,new_jury_reward)
+    await modDB.updateJuryRewardforRequests(pollObject.id,new_jury_reward)
+
+}
+
+function handleNoDueDatePoll(){
+     // 
+}
+
+cron.schedule('*/60 * * * * *', closeFinishedPolls);
```

Comment:
does this work? https://www.npmjs.com/package/node-cron#allowed-values says allowed values are 0-59. If it works keep it.

Reply:
I wanted to discuss the routine time intervals with you so they are open to change